### DISCUSSION TITLE
AM2301 change HumOffset + - 10  to  + - 45

### DISCUSSION THREAD

**Discussion by boy0038** [arendst/Tasmota#10317] [Q&A] [answered]
Hi there

I have AM2301 with Humidity error of 42 difference with HumOffset 0 but it only works up to + -10 difference
you can change that except in a script so that it is also displayed correctly with the IP from the sonoff

translated with Google

thanks

**Comment by sfromis** [comment, marked as answer]
With such a huge difference, I'd consider it badly configured, or just headed for the trash bin

**Reply by g6094199** [reply]
that was nor the question of the op nor its the solution.  

plz answer the question why hum offset is limitied to only +/-10%. +/-10% way not enough and within the error of some sensors.

**Reply by sfromis** [reply]
It was an observation of the wanted approach not being a reasonable way to deal a hardware issue of 42 difference. And the AM2301 is **really** not a good sensor anyway. Just do not expect good humidity readings.

The offset is intended for calibration tweaks, not for totally bonkers values.

One technical factor is that the offset is stored as a signed 8-bit number.

**Comment by boy0038** [comment]
the temperature just doesn't match the humidity
that the sensor is defective
have larger and different with several AM2301

**Reply by sfromis** [reply]
Quite often, what's labled as Sonoff AM2301 has a different sensor inside. I'd try configuring as SI7021 instead.

**Comment by boy0038** [comment]
the humidity is the same

**Comment by boy0038** [comment]
if the weather is right i will change the sensor

**Comment by spcqike** [comment]
I just want to reopen this as I face a similar problem. 

I have a BME680 which reads a to high temperature. I need to adjust the temperature by -5.7 degrees.

So far so good. Bad thing is that this also changes the humidity. Or it should. the Tempoffset doesn’t change the humidity itself. But -10…10 isn’t enough to compensate such wrong temperatures. 

E.g. need to use -5.7 degrees and +19%. 

It would be good, if Tempoffset would also change the humidity, as relativ humidity depends on temperature. 
https://rechneronline.de/barometer/luftfeuchtigkeit.php

if I insert my values (25C, 50%) and change it to 20C, I get 66.6% just because of the lower temperature. My (hopefully) real values are 20C and 69%. 

I also tried another BME680 from another manufacturer, same readings. (Also also read that this can be normal as BMEs don’t come Hardware calibrated … but I’m not to sure about that) 

so, my question: is it possible to increase the Humoffset? Or, is it possible to let Tempoffset also change humidity accordingly? In the later case, -10…10 Humoffset would be enough for fine tuning.

**Reply by sfromis** [reply]
With such bad values, something is really wrong, and wanting a excessively big offsets is not a solution. That's not how calibration works. You may have placed the sensor too close to some electronics producing heat. Some "thermal distance" is important.

**Reply by spcqike** [reply]
I don’t see anything warm, except the wemos d1. But this shouldn’t heat the sensor up by 5K. Also, I have the same readings with two different BME680. And two other thermostats are lower and have a higher humidity (which would match other sensors around the house) 

i will try a BME280 and also some jumper wires to get the sensor a little further away. 

Still, I think It would be a good thing to Let the Tempoffset also change the humidity. As relative humidity always depends on temperature. (And pressure, but not as strong) 
If we know, that the temperature is wrong, the humidity must be wrong, too. And, as warmer air can hold more water, the fixed offset for humidity isn’t useful. 
instead it would be good to first calculate the absolute humidity (out of sensors temperature and humidity), offset the temperature and calculate the new humidity out of the changed temperature and the absolute humidity. 

One can see this here 
https://planetcalc.com/2167/


I guess, if the bme280 also has wrong readings, I will just use node red or so to recalculate more realistic values for this device :) 

edit:
It looks like this is a more or less common problem with these sensors and there are possibilities to compensate this 
https://github.com/pimoroni/bme680-python/issues/11

**Reply by sfromis** [reply]
The D1 Mini is a radiator which can generate a significant temperature lift, if placed in a bad way relative to a temperature sensor.

**Reply by g6094199** [reply]
i have the same with a BME280, which is off to far. its far from being optimal mounted inside a floor heating thermostat with backlite, which heats up the sensor unnecessary. we all know hum is depends on temp. also maybe its a fake, cause its brand new from china. the only chance to the the values into range is  a bigger offset than +/-10. **alt least +/-25**

i'm paying my rent with sensors stuff so i know thats far away from beeing a perfect solution but i dont want to have a ugly thermostat with holes wires and stuff hanging around so everything need to be inside the original case..

**Reply by sfromis** [reply]
You cannot use a simple offset to "correct" a bad humidity reading due to self-heating. That's not how humidity in air works, as the relationship between temperature and *relative* humidity (with the same absolute humidity) is not an offset, but varies with temperature, and not even linear. You need a better formula than expecting a simple addition to fix the number. Such a formula would also need the environmental temperature which the humidity is expected to match.

**Comment by sfromis** [comment]
Anyway, if you want to convert bad numbers to be bad in a different way (or maybe even better), here's an example of using rules to pick up a sensor value, modify it, and then publish an additional payload with the changed value.
https://tasmota.github.io/docs/Rules/#adjust-a-value-and-send-it-over-mqtt
